app/controllers/SecureMessagingController.scala

```diff
@@ -34,24 +36,46 @@ class SecureMessagingController @Inject()(
   messageConnector: SecureMessageFrontendConnector,
   mcc: MessagesControllerComponents,
   inbox_wrapper: inbox_wrapper,
-  conversation_wrapper: conversation_wrapper
+  conversation_wrapper: conversation_wrapper,
+  reply_result_wrapper: reply_result_wrapper
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
-  def displayInbox: Action[AnyContent] = (authenticate andThen verifiedEmail andThen secureMessagingFeatureAction).async { implicit req =>
+  val actions = authenticate andThen verifiedEmail andThen secureMessagingFeatureAction
+
+  val displayInbox: Action[AnyContent] = actions.async { implicit request =>
     messageConnector
       .retrieveInboxPartial()
       .map { partial =>
         Ok(inbox_wrapper(HtmlFormat.raw(partial.body)))
       }
   }
 
-  def displayConversation(client: String, conversationId: String): Action[AnyContent] =
-    (authenticate andThen verifiedEmail andThen secureMessagingFeatureAction).async { implicit req =>
-      messageConnector
-        .retrieveConversationPartial(client, conversationId)
-        .map { partial =>
-          Ok(conversation_wrapper(HtmlFormat.raw(partial.body)))
-        }
+  def displayConversation(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    messageConnector
+      .retrieveConversationPartial(client, conversationId)
+      .map(partial => Ok(wrapperFormForPartial(partial)))
+  }
+
+  def displayReplyResult(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    messageConnector
+      .retrieveReplyResult(client, conversationId)
+      .map(partial => Ok(reply_result_wrapper(HtmlFormat.raw(partial.body))))
+  }
+
+  def submitReply(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    request.body.asFormUrlEncoded.map { reply =>
+      messageConnector.submitReply(client, conversationId, reply)
     }
+
+    // For the time being... until the downstream service is ready
+    Future(Redirect(routes.SecureMessagingController.displayReplyResult(client, conversationId)))
+
+  // Future.successful(NoContent)
+  }
+
+  private def wrapperFormForPartial(partial: ConversationPartial)(implicit request: Request[_]): HtmlFormat.Appendable = {
+    val csrfToken = CSRF.getToken.get.value
```

Comment:
Please don't use `get` on `Option` in production code.

Reply:
This is not production code, it is just temporary. We won't need to do this when DC is ready. They will also provide the CSRF Token that we do not have currently in the prototype.